### ISSUE TITLE
unable to make nested target when source is not nested

### ISSUE THREAD

**Issue by horijon** [perpk/json-xform#42]
Want to convert/unflatten from
```
const jsonData = {
    "description": "SARS-CoV-2 (COVID-19) IgM Ab [Units/volume] in Serum or Plasma by Immunoassay"
};
```
to result
```
{
   "category":[
      {
         "coding":[
            {
               "display":"SARS-CoV-2 (COVID-19) IgM Ab [Units/volume] in Serum or Plasma by Immunoassay"
            }
         ]
      }
   ]
}
```

How do i write mapping logic for such case? 
@perpk I am eagerly seeking for your kind help! And thank you for the library!

**Comment by perpk** [comment]
Hi @horijon and thanks for the inquiry!

You can define nesting in your target object by using chaining:
```
const template = {
    fieldset: [{
        from: "description",
        to: "category.coding.display"
    }]
};
```
This will create the result:

```
{
  category: {
    coding: {
      display: "SARS-CoV-2 (COVID-19) IgM Ab [Units/volume] in Serum or Plasma by Immunoassay"
    }
  }
}
```

Yet, it is not exactly what you asked for, as you need "category" and "coding" to be both arrays. Unfortunately, there is no way this can be done in the current version. You can nest, though the type in the result isn't determined automatically to array nor there's a way to do such per definition in the template. There is an active issue #28 which will picked up soon and will allow the user to define an array as the type for a particular mapping.

In case you need this feature (#28 _to have arrays defined for the outcome per template_) urgently, please let me know so I can start working on it earlier than scheduled.

**Comment by horijon** [comment]
Thank you @perpk for your quick response. I am trying to work on it like from a week. In addition to concatenation (for now can be achieved using withTemplate), I also need some transformation logic eg; format date.
I am looking for many libraries and i came to yours which seems simple and nice. But to use it at least i need these features to work 1) unflatten to array/complex structures 2) transformation logics (may be js functions) for the fields (at least concatenation and formatDate).

I am eagerly looking for such solution. The sooner the better for me. And again thank you a lot.

**Comment by perpk** [comment]
Hi @horijon, thanks for the details. I'll look into possible implementations asap.

**Comment by perpk** [comment]
Hi @horijon ! I just released [v1.2.0](https://www.npmjs.com/package/@perpk/json-xform). Both, defining arrays in the output and date formatting is now available and documented. Please contact me in case you require any assistance.

Thanks!
